addon/utils/store.js

```diff
@@ -21,11 +21,30 @@ export default EmberObject.extend({
   },
 
   initialsFor(properties) {
-    let key = hash(properties);
+    let hashableProp = this._ensureHashable(properties);
+    let key = md5(hashableProp);
     return this.get('cache')[key] || this._create(key, properties);
   },
 
   _create(key, properties) {
     return this.get('cache')[key] = this.get('generator').generate(properties);
+  },
+
+  _ensureHashable(properties) {
+    if (typeOf(properties) !== 'object') {
+      // this should never happen, but just in case
+      return properties.toString();
+    }
+
+    let pKeys = Object.keys(properties).sort();
+    let result = [];
+    pKeys.forEach((key) => {
```

Comment:
What about `.map` here?

Reply:
I didn't do a `.map` because I wasn't positive how the nested check would work in that case.  I didn't want to end up with `[a, b, [c, d]]` so chose to concat instead.  Open to that being incorrect